micro-rdk/src/common/robot.rs

```diff
@@ -242,6 +275,31 @@ impl LocalRobot {
             components.map_err(RobotError::RobotParseConfigError)?,
             registry,
         )?;
+
+        // TODO: When cfg's on expressions are valid, remove the outer scope.
+        #[cfg(feature = "data")]
+        {
+            match DataManager::<StaticMemoryDataStore>::from_robot_and_config(&robot, config_resp) {
+                Ok(Some(mut data_manager)) => {
+                    let _ = robot
+                        .data_manager_sync_task
+                        .insert(Box::new(data_manager.get_sync_task()));
+
+                    let _ = robot
```

Comment:
There is currently a clippy error here that I don't fully understand:
```
error: non-binding `let` on a future
   --> micro-rdk/src/common/robot.rs:289:21
    |
289 | / ...   let _ = robot
290 | | ...       .data_manager_collection_task
291 | | ...       .insert(robot.executor.spawn(async move {
292 | | ...           if let Err(err) = data_manager.data_collection_task().await {
293 | | ...               log::error!("error running data manager: {:?}", err)
294 | | ...           }
295 | | ...       }));
    | |______________^
    |
    = help: consider awaiting the future or dropping explicitly with `std::mem::drop`
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#let_underscore_future
    = note: `-D clippy::let-underscore-future` implied by `-D warnings`
    = help: to override `-D warnings` add `#[allow(clippy::let_underscore_future)]`

error: could not compile `micro-rdk` (lib) due to previous error
```

Reply:
@mattjperez and I figured this out. Using `insert` here was not quite right, using `replace` placated clippy.